cmd/internal/ecs/add.go

```diff
@@ -870,6 +879,121 @@ func modifyTaskState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowS
+	}
+
+	return awf_next(waitForRestartState)
+}
+
+func waitForRestartState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowState], err error) {
+	reportStep("Wait for ECS Service")
+	startTime := time.Now()
+	endTime := startTime.Add(5 * time.Minute)
+
+	failure := errors.New("Cannot determine whether service was successfully upgraded. Please check the AWS console.\n")
+
+	// Check that there's a new deployment in the service
+	deploymentID, deployment, err := wf.GetDeploymentMatchingTask(wf.ecsServiceARN)
+	if err != nil {
+		if errors.Is(err, noDeploymentFound) {
+			printer.Errorf("Could not locate a deployment for the new task definition \"%s:%s\" (%s).\n",
+				wf.ecsTaskDefinitionFamily,
+				wf.ecsTaskDefinition.Revision,
+				wf.ecsTaskDefinitionARN)
+		} else {
+			printer.Errorf("Error checking service state: %v\n", err)
+		}
+		return awf_error(failure)
+	}
+	printer.Infof("Found new deployment with ID %q; its state is %s.\n", deploymentID, deployment.RolloutState)
+
+	if deployment.RolloutState == types.DeploymentRolloutStateInProgress {
+		printer.Infof("Waiting for the deployment to reach COMPLETED.\n")
+	}
+
+	ticker := time.NewTicker(15 * time.Second)
+	for deployment.RolloutState == types.DeploymentRolloutStateInProgress {
+		select {
+		case t := <-ticker.C:
+			if t.After(endTime) {
+				printer.Warningf("Giving up after five minutes.\n")
+				if deployment.RunningCount > 0 {
+					printer.Infof("Some tasks did start, indicating that the new task definition is OK.\n")
+				}
+				return awf_error(failure)
+			}
+			deployment, err = wf.GetDeploymentByID(wf.ecsServiceARN, deploymentID)
+			if err != nil {
+				printer.Warningf("Error checking service state: %v", err)
+				continue
+			}
+			duration := time.Now().Sub(startTime)
+			printer.Infof("%02d:%02d after starting deployment %q: %d failed tasks, %d running tasks, %d pending tasks, overall status %s/%s.\n",
+				duration/time.Minute, (duration%time.Minute)/time.Second,
+				deploymentID, deployment.FailedTasks, deployment.RunningCount, deployment.PendingCount,
+				aws.ToString(deployment.Status), deployment.RolloutState)
+
+			if deployment.RolloutState == types.DeploymentRolloutStateFailed {
+				printer.Errorf("Deployment of the new task definition failed. The reason given by AWS is: %q",
+					aws.ToString(deployment.RolloutStateReason))
+				// TODO: guide the user through this?
+				printer.Infof("You can use the AWS web console to delete the task definition \"%s:%s\". The previous task definition should still be in use.\n",
+					wf.ecsTaskDefinitionFamily, wf.ecsTaskDefinition.Revision)
+				return awf_error(errors.New("The modified task failed to deploy. Please contact support@akitasoftware.com for assistance."))
+			}
+		}
+	}
+
+	reportStep("ECS Service Updated")
+	printer.Infof("Deployment successful! Please return to the Akita web console.\n")
```

Comment:
Provide a link?

Reply:
I thought about it, but what we expect is that they have come from the onboarding workflow and just need to find the tab they had open.  If @paul-russo can suggest a route to use, though, I'd be happy to include one.

Reply:
Makes sense. We can add a link later if it turns out that people are falling off here.